dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -487,6 +491,108 @@ public Map<String, Object> execute(User loginUser, Integer workflowInstanceId, E
         return execute(loginUser, processDefinition.getProjectCode(), workflowInstanceId, executeType);
     }
 
+    /**
+     * do action to execute task in process instance
+     *
+     * @param loginUser login user
+     * @param projectCode project code
+     * @param processInstanceId process instance id
+     * @param startNodeList start node list
+     * @param taskDependType task depend type
+     * @return execute result code
+     */
+    @Override
+    public Map<String, Object> executeTask(User loginUser, long projectCode, Integer processInstanceId,
```

Comment:
Please don't use Map<String,Object> as result object, because it's not clear enough. It's better to throw a ServiceException if not success. 
```suggestion
    public void executeTask(User loginUser, long projectCode, Integer processInstanceId,
```